borg/archive.py

```diff
@@ -503,7 +501,10 @@ def process_file(self, path, st, cache):
             else:
                 self.hard_links[st.st_ino, st.st_dev] = safe_path
         path_hash = self.key.id_hash(os.path.join(self.cwd, path).encode('utf-8', 'surrogateescape'))
+        first_run = not cache.files
         ids = cache.file_known_and_unchanged(path_hash, st)
+        if first_run:
+            logger.info('processing files')
```

Comment:
if the reason why this is emitted is to tell the user that it is now actually working on the files (if the user does not see that anyway because verbose is not set):  should this be done in do_create or some other higher level function, before it dives into the recursion? you would not need the first_run hack there.